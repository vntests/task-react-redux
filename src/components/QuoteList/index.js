import React from 'react';
import PropTypes from 'prop-types';

import { Button } from 'reactstrap';

import Quote from '../Quote';

import './index.css';

const QuoteList = ({ quoteList, getQuote, removeQuote }) => (
  <div>
    {quoteList && quoteList.map(quote => (
      <Quote
        key={quote.ID}
        title={quote.title}
        content={quote.content}
        link={quote.link}
        remove={() => { removeQuote(quote) }}
      />
    ))}
    <Button
      type="button"
      block
      color="primary"
      onClick={getQuote}
    >
      Get one more quote
    </Button>
  </div>
);

QuoteList.propTypes = {
  quoteList: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string,
    content: PropTypes.string,
    link: PropTypes.string
  })),
  getQuote: PropTypes.func,
  removeQuote: PropTypes.func
};

export default QuoteList;
