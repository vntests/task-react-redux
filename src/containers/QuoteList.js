import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { getQuoteList } from '../actions/quote';

import QuoteList from '../components/QuoteList';

class QuoteListContainer extends Component {
  static propTypes = {
    quoteList: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string,
      content: PropTypes.string,
      link: PropTypes.string
    })),
    getQuoteList: PropTypes.func
  };

  render() {
    const { quoteList } = this.props;

    return (
      <QuoteList
        quoteList={quoteList}
      />
    );
  }

  componentDidMount() {
    const { getQuoteList } = this.props;

    getQuoteList({
      orderby: 'rand',
      postsPerPage: 1
    });
  }
}

const mapStateToProps = state => ({
  quoteList: state.quoteList
});

const mapDispatchToProps = dispatch => ({
  getQuoteList: options => {
    dispatch(getQuoteList(options))
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(QuoteListContainer);
