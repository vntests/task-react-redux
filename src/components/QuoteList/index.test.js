import React from 'react';
import renderer from 'react-test-renderer';

import QuoteList from './';

jest.mock('../Quote', () => 'Quote');

test('<QuoteList /> should render', () => {
  const component = renderer.create(
    <QuoteList />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('<QuoteList /> should render quote list with N quotes', () => {
  const component = renderer.create(
    <QuoteList
      quoteList={[
        {
          ID: 1,
          title: 'Title',
          content: 'Content',
          link: 'Link'
        }
      ]}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
