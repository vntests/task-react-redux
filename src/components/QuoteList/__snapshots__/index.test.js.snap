// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuoteList /> should render 1`] = `<div />`;

exports[`<QuoteList /> should render quote list with N quotes 1`] = `
<div>
  <Quote
    content="Content"
    link="Link"
    title="Title"
  />
</div>
`;
